Model/Customer.php

```diff
@@ -0,0 +1,167 @@
+     * @var \Magento\Customer\Model\ResourceModel\Customer
+     */
+    protected $magentoCustomerResource;
+
+    /**
+     * @var \Magento\Customer\Model\Session
+     */
+    protected $magentoCustomerSession;
+
+    /**
+     * @var \Omise\Payment\Model\Omise
+     */
+    protected $omise;
+
+    /**
+     * @var \Omise\Payment\Model\Api\Customer
+     */
+    protected $customerAPI;
+
+    public function __construct(
+        MagentoCustomerResource $magentoCustomerResource,
+        MagentoCustomerSession  $magentoCustomerSession,
+        Omise                   $omise,
+        OmiseCustomerAPI        $customerAPI
+    ) {
+        $this->magentoCustomerResource = $magentoCustomerResource;
+        $this->magentoCustomerSession  = $magentoCustomerSession;
+        $this->magentoCustomer         = $this->magentoCustomerSession->getCustomer();
+        $this->omise                   = $omise;
+        $this->customerAPI             = $customerAPI;
+
+        $this->omise->defineUserAgent();
+        $this->omise->defineApiVersion();
+        $this->omise->defineApiKeys();
+
+        $this->initializeObject();
+    }
+
+    /**
+     * Check whether the executor is logged in or not.
+     *
+     * @return bool
+     */
+    public function isLoggedIn()
+    {
+        return $this->getMagentoCustomerId() ? true : false;
+    }
+
+    /**
+     * Get an Omise customer id that belongs to
+     * a specific user who execute the method.
+     *
+     * @return string
+     */
+    public function getId()
+    {
+        return $this->magentoCustomer->getData(self::OMISE_CUSTOMER_ID_FIELD);
+    }
+
+    /**
+     * Get a Magento customer id that belongs to
+     * a specific user who execute the method.
+     *
+     * @return int|null
+     *
+     * @see    magento/module-customer/Model/Session.php
+     */
+    public function getMagentoCustomerId()
+    {
+        return $this->magentoCustomerSession->getCustomerId();
+    }
+
+    /**
+     * Create an Omise customer, then assigning its Omise customer id
+     * back to the customer's attribute who execute the method.
+     *
+     * @return self
+     */
+    public function create(array $params)
+    {
+        $this->customerAPI = $this->customerAPI->create($params);
+
+        $this->magentoCustomer->setData(self::OMISE_CUSTOMER_ID_FIELD, $this->customerAPI->id);
+        $this->magentoCustomerResource->saveAttribute($this->magentoCustomer, self::OMISE_CUSTOMER_ID_FIELD);
+
+        return $this;
+    }
+
+    /**
+     * Attach an Omise Token to a Customer object.
+     *
+     * @param string $cardToken
+     *
+     * @return self
+     */
+    public function addCard($cardToken)
+    {
+        if (! $this->getId()) {
+            $this->create(array(
+                'email'       => $this->magentoCustomer->getEmail(),
+                'description' => trim($this->magentoCustomer->getFirstname() . ' ' . $this->magentoCustomer->getLastname())
+            ));
+        }
```

Comment:
Should this be in here? Smells like something that should be elsewhere

Reply:
Uhmm feel a bit off about this too.